packages/components/src/Button/Button.tsx

```diff
@@ -27,7 +27,7 @@ interface ButtonFoundationProps {
   readonly label?: string;
   readonly loading?: boolean;
   readonly size?: "small" | "base" | "large";
-  onClick?(): void;
+  onClick?(event: React.MouseEvent<HTMLElement>): void;
```

Comment:
I'm just going to add this here, but applies to all of the work. Is there a reason you chose to pass the event as a generic `HTMLElement` instead of being more specific.

```tsx
onClick?(event: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>): void
```

Reply:
@eddysims I followed the example in the card component, however I'm okay with making the change to make it more specific. 

Do you think anything is gained by making it more specific, and is the added work to maintain that if we change components worth it?

Reply:
Was looking in to this yesterday and it looks like we gain a little bit more data from the event (and by that I mean typescript would allow us). I.e. the `HTMLButtonElement` gives you `disabled` to check if the button is disabled or not. An `a` tag to grab the `href`.

I'm okay with keeping it generic and making it specific when someone does need to as we're starting with the generic type anyway.

Reply:
Sounds good - I also chose HTMLElement as everything would be a child of this type, but I can explicitly change the onClick to be specific for all the cases. 🔜